model/src/test/java/org/projectnessie/model/TestNamespace.java

```diff
@@ -85,19 +85,19 @@ public void testOneElement() {
   public void testRoundTrip() {
     List<String> elements = asList("a", "b.c", "namespace");
     String pathString = UriUtil.toPathString(elements);
-    String expectedPathString = "a.b\u0000c.namespace";
+    String expectedPathString = "a.b%00c.namespace";
     assertThat(pathString).isEqualTo(expectedPathString);
     Namespace namespace = Namespace.parse(pathString);
     assertThat(namespace.name()).isEqualTo(pathString);
     assertThat(namespace.getElements()).isEqualTo(elements);
     assertThat(namespace.toString()).isEqualTo(pathString);
     assertThat(namespace.toPathString()).isEqualTo(pathString);
     assertThat(namespace.name()).startsWith("a.b");
-    assertThat(namespace.name()).startsWith("a.b\u0000");
-    assertThat(namespace.name()).startsWith("a.b\u0000c");
-    assertThat(namespace.name()).startsWith("a.b\u0000c.namespa");
+    assertThat(namespace.name()).startsWith("a.b%00");
+    assertThat(namespace.name()).startsWith("a.b%00c");
+    assertThat(namespace.name()).startsWith("a.b%00c.namespa");
```

Comment:
Side note: the assertThat...startsWith repetition is weird - the last one should be sufficient

Reply:
this specifically tests that zero bytes are correctly taken into account, since we're using `startsWith` logic in the namespace API